ci/scripts/rust_clippy.sh

```diff
@@ -18,6 +18,4 @@
 # under the License.
 
 set -ex
-cargo clippy --all-targets --workspace --features avro,pyarrow,integration-tests -- -D warnings
-cd datafusion-cli
-cargo clippy --all-targets --all-features -- -D warnings
```

Comment:
Not being able to clippy with `--all-features` is a loss. 

I know there are few mutually-excluding features (sigh), but we can do better than clippying 3 features and not clippying all the rest.

Let's follow-up on this.

In a follow-up we should also clippy (or check) with no-default-features.

Reply:
I agree. We could change this to:
```
cargo clippy --workspace --all-targets --all-features --exclude datafusion-benchmarks -- -Dwarnings
cargo clippy -p datafusion-benchmarks --all-targets -- -Dwarnings
```
What do you think?

Reply:
i think all-features may be problematic for core, not benchmarks. but if the abve works, great!
let's do this in a follow-up though.
let's focus on getting cargo.lock into the repo, and do non-mandatory clippy improvement on its own focused pr

(review time is non-linear of its size)

Reply:
The above passes, except for one warning:
```
error: use of a disallowed method `tokio::task::spawn`
  --> datafusion/sqllogictest/src/engines/postgres_engine/mod.rs:93:27
```
I think we can just add `#[allow(clippy::disallowed_methods)]` there.

But I agree, let's do this in a follow-up.